src/components/StyleTitle/StyleTitle.tsx

```diff
@@ -0,0 +1,15 @@
+import { useMemo } from "react";
+
+type StyleCardProps = CustomStyle;
+
+export const StyleTitle = ({ title, node }: StyleCardProps) => {
+  const slug = useMemo(() => {
+    return title.toLowerCase().replace(/ /g, "-");
+  }, [title]);
```

Comment:
Think it would be wise to make this a lib function so you can reuse it.
Additionally, let's make this a bit more robust:
```suggestion
  const slug = useMemo(() => slugify(title), [title]);
```

```js
const slugify = (string) => string
  .normalize('NFD')
  .toLowerCase()
  .replace(/[^a-z0-9 ]/g, '')
  .replace(/\s+/g, '-')

```

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize